flytekit/models/sagemaker/parameter_ranges.py

```diff
@@ -0,0 +1,237 @@
+        """
+
+        if self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.AUTO:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.AUTO
+        elif self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.LINEAR:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.LINEAR
+        elif self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.LOGARITHMIC:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.LOGARITHMIC
+        elif self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.REVERSELOGARITHMIC:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.REVERSELOGARITHMIC
+        else:
+            raise _user_exceptions.FlyteValidationException("Invalid SageMaker HyperparameterScalingType specified")
+
+        return _idl_parameter_ranges.ContinuousParameterRange(
+            max_value=self._max_value,
+            min_value=self._min_value,
+            scaling_type=scaling_type,
+        )
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object):
+
+        scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.AUTO
+        if pb2_object.type == _idl_parameter_ranges.HyperparameterScalingType.LINEAR:
+            scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.LINEAR
+        elif pb2_object.type == _idl_parameter_ranges.HyperparameterScalingType.LOGARITHMIC:
+            scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.LOGARITHMIC
+        elif pb2_object.type == _idl_parameter_ranges.HyperparameterScalingType.REVERSELOGARITHMIC:
+            scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.REVERSELOGARITHMIC
+
+        return cls(
+            max_value=pb2_object.max_value,
+            min_value=pb2_object.min_value,
+            scaling_type=scaling_type,
+        )
+
+
+class IntegerParameterRange(_common.FlyteIdlEntity):
+    def __init__(
+            self,
+            max_value: int,
+            min_value: int,
+            scaling_type: _sdk_sagemaker_types.HyperparameterScalingType,
+    ):
+        """
+        :param int max_value:
+        :param int min_value:
+        :param _sdk_sagemaker_types.HyperparameterScalingType scaling_type:
+        """
+        self._max_value = max_value
+        self._min_value = min_value
+        self._scaling_type = scaling_type
+
+    @property
+    def max_value(self):
+        """
+        :return: int
+        """
+        return self._max_value
+
+    @property
+    def min_value(self):
+        """
+
+        :return: int
+        """
+        return self._min_value
+
+    @property
+    def scaling_type(self):
+        """
+        enum value from HyperparameterScalingType
+        :return: int
+        """
+        return self._scaling_type
+
+    def to_flyte_idl(self):
+        """
+        :return: _idl_parameter_ranges.IntegerParameterRange
```

Comment:
if using type comments, let's stick with `rtype` since that's the pattern matt established.